### ISSUE TITLE
TLS Connection fails when two URLs present

### ISSUE THREAD

**Issue by oliverhausler** [nats-io/nats.java#279]
If I give `conntect()` two `tls://` URLs as array, the connection fails with `TLS Handshake Error: EOF`, with only one of the URLs (either one) it connect to the cluster.

My suspection is that the Java client does not try round-robin and instead starts both connections at the same time, but only a guess; or that non-TLS communication mixes with TLS communication.

Two non-TLS URLs work fine.

**Comment by sasbury** [comment]
It does them one at a time, so there is possible an issue with how the server is handling TLS vs how the client is (i saw something elsewhere that made me think that anyway)

**Comment by oliverhausler** [comment]
I can reproduce this consistently. When I pass 2 URLs, connecting almost always fails (maybe 1 out of 100 succeeds). When I pass only one URL, connecting always succeeds.

**Comment by wutkemtt** [comment]
Hello, 
we are having the same problem and can reproduce it too.
Is there any update ?

**Comment by gstaware** [comment]
We also are experiencing same problem. Are we looking to fix this issue?

**Comment by gstaware** [comment]
We used benchmark code and below is the error after trying with -s tls://server1:4222,tls://server2:4222

Connection URL null
An exception occurred...
java.util.concurrent.ExecutionException: java.io.IOException: SSL required by server.
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:206)
        at io.nats.client.impl.NatsConnection.tryToConnect(NatsConnection.java:455)
        at io.nats.client.impl.NatsConnection.connect(NatsConnection.java:217)
        at io.nats.client.impl.NatsImpl.createConnection(NatsImpl.java:29)
        at io.nats.client.Nats.createConnection(Nats.java:271)
        at io.nats.client.Nats.connect(Nats.java:186)
        at io.nats.examples.benchmark.NatsBench$PubWorker.run(NatsBench.java:420)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: SSL required by server.
        at io.nats.client.impl.NatsConnection.upgradeToSecureIfNeeded(NatsConnection.java:557)
        at io.nats.client.impl.NatsConnection$1.call(NatsConnection.java:441)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        ... 1 more

**Comment by oliverhausler** [comment]
Is this going to be fixed?

**Comment by RichardHightower** [comment]
We were able to repeat this issue after a few attempts. We configured the bridge to w/ multiple TLS connected servers and failover worked fine. This is an issue with the benchmark utility which always takes the last argument as the subject. This means if you do not provide a stream name, it will take the tls argument as the stream name, and then you will get the TLS error mentioned. We tested the benchmark and got it to work. Then we tried to use the same args listed above and got the same message. Solution: add a subject as the final argument to the benchmark utility.

**Comment by RichardHightower** [comment]
```static final String usageString = "\nUsage: java NatsBench [-s server] [-tls] [-np num] [-ns num] [-n num] [-ms size] "
        + "[-csv file] <subject>\n\nOptions:\n"
        + "    -s  <urls>                     The nats server URLs (comma-separated), use tls:// or opentls:// to require tls\n"
        + "    -np <int>                       Number of concurrent publishers (1)\n"
        + "    -ns <int>                       Number of concurrent subscribers (0)\n"
        + "    -n  <int>                       Number of messages to publish (100,000)\n"
        + "    -ms <int>                       Size of the message (128)\n"
        + "    -csv                            Print results to stdout as csv (false)\n"
        + "    -tls                            Set the secure flag on the SSL context to true (false)\n"
        + "    -stats                          Track and print out internal statistics (false)\n";```

If we pass these args it runs fine... 
`{"-s","tls://localhost:4222,tls://localhost:4223,tls://localhost:4224", "-tls","test01"};`

if we leave off the subject 


`{"-s","tls://localhost:4222,tls://localhost:4223,tls://localhost:4224", "-tls"};`

The utitliy grabs -tls as the subject and we get the same error message. We did extensive testing with the bridge and the SSL support and cluster support for NATS. It works as expected. Then we turned our attention to the benchmark utility itself. And found this behavior.

**Comment by oliverhausler** [comment]
@RichardHightower It fails for me from Java code. Reopen?

```
new NatsClient().connect("name", urls); // URLs is a HashSet<String>
```

**Comment by gstaware** [comment]
We verified this and it is working fine now. We needed to pass -tls flag in benchmarks utility code as suggested by Rick. Thanks @RichardHightower @ColinSullivan1

**Comment by oliverhausler** [comment]
@RichardHightower This hasn't been released into production yet. Is there an expected ETA after 2.8.0?

**Comment by oliverhausler** [comment]
@RichardHightower This hasn't been released into production yet. Is there an expected ETA after 2.8.0?

**Comment by danpoland** [comment]
This is not fixed in 2.10.0.
